kratos/future/solving_strategies/schemes/static_scheme.h

```diff
@@ -0,0 +1,437 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//                   Riccardo Rossi
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/csr_matrix.h"
+#include "containers/system_vector.h"
+#include "includes/model_part.h"
+#include "includes/kratos_parameters.h"
+#include "spaces/kratos_space.h"
+#include "utilities/builtin_timer.h"
+#include "utilities/dof_utilities/dof_array_utilities.h"
+#include "utilities/entities_utilities.h"
+#include "utilities/openmp_utils.h" //TODO: SOME FILES INCLUDING scheme.h RELY ON THIS. LEAVING AS FUTURE TODO.
+#include "utilities/parallel_utilities.h"
+#include "utilities/timer.h"
+
+#ifdef KRATOS_USE_FUTURE
+#include "future/linear_solvers/amgcl_solver.h"
+#include "future/solving_strategies/schemes/implicit_scheme.h"
+#include "future/solving_strategies/builders/builder.h"
+#endif
+
+namespace Kratos::Future
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief Implicit scheme TLS type definition
+ * Thread Local Storage container to be used in the parallel assembly of implicit problems
+ * @tparam DataType data type of the problem to be solved
+ */
+template<class TDataType = double >
+struct StaticThreadLocalStorage
+{
+    // Local LHS contribution
+    DenseMatrix<TDataType> LocalMatrix;
+
+    // Local RHS constribution
+    DenseVector<TDataType> LocalVector;
+
+    // Vector containing the localization in the system of the different terms
+    Element::EquationIdVectorType LocalEqIds;
+
+    // Vector containing the slave equation ids
+    MasterSlaveConstraint::EquationIdVectorType SlaveEqIds;
+
+    // Vector containing the master equation ids
+    MasterSlaveConstraint::EquationIdVectorType MasterEqIds;
+};
+
+/**
+ * @class StaticScheme
+ * @ingroup KratosCore
+ * @brief This class provides the implementation of the static scheme
+ * @author Ruben Zorrilla
+ */
+template<class TSparseMatrixType, class TSystemVectorType, class TSparseGraphType>
+class StaticScheme : public ImplicitScheme<TSparseMatrixType, TSystemVectorType, TSparseGraphType>
+{
+public:
+
+    // FIXME: Does not work... ask @Charlie
```

Comment:
Ping @roigcarlo

Reply:
It is the same registry issue I commented [here](https://github.com/KratosMultiphysics/Kratos/pull/13762#discussion_r2309334184).